### PR TITLE
Create details_cubit.dart, details_item.dart, details_page.dart, deta…

### PR DESCRIPTION
**Pull request by Alexandr019**
…ils_repo.dart, navigation.dart. Update food_page.dart, main.dart, food_item.dart.

### CODE REVIEW

File: lib/details_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  DetailsCubit(this._detailsRepo) : super(DetailsState());
+
+  void fetchDetails(String globalId) async {
+    emit(DetailsState(isLoading: true));
+    List<DetailsItem> details = await _detailsRepo.fetchDetails(globalId);
+    emit(DetailsState(details: details));
+  }
+}
+
+class DetailsState {
```
Comment (Oliinyk-Artemii) [inline lib/details_cubit.dart:17]: add equatable lib

**Review by Oliinyk-Artemii — CHANGES_REQUESTED** [review]

File: lib/details_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import 'package:food/details_repo.dart';
+import 'package:food/details_item.dart';
+
+class DetailsCubit extends Cubit<DetailsState> {
+  final DetailsRepo _detailsRepo;
+
+  DetailsCubit(this._detailsRepo) : super(DetailsState());
+
+  void fetchDetails(String globalId) async {
+    emit(DetailsState(isLoading: true));
```
Comment (Oliinyk-Artemii) [inline lib/details_cubit.dart:11]: I would implement copyWith method and would change values via this method

File: lib/details_page.dart [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  late DetailsCubit _detailsCubit;
+
+  @override
+  void initState() {
+    super.initState();
+    setState(
+      () {
+        _detailsCubit = DetailsCubit(
+          DetailsRepo(
+            Dio(
```
Comment (Oliinyk-Artemii) [inline lib/details_page.dart:33]: We need to use DI (dependency injection) here. Remember me plz later, I'll show you how to implement it and why we need this

Comment (Alexandr019) [reply lib/details_page.dart]: done

**Pull request merged**